Algorithm/QCAlgorithm.cs

```diff
@@ -1528,7 +1534,8 @@ public Future AddFuture(string ticker, Resolution? resolution = null, string mar
         {
             if (market == null)
             {
-                if (!BrokerageModel.DefaultMarkets.TryGetValue(SecurityType.Future, out market))
+                if (!SymbolPropertiesDatabase.TryGetMarket(ticker, SecurityType.Future, out market)
+                    && !BrokerageModel.DefaultMarkets.TryGetValue(SecurityType.Future, out market))
```

Comment:
Is there a reason why we chose not to override the `market` parameter if it equals `Market.USA`? 

When we go to fetch the entry from the MHDB, it will inevitably fail even if we let the user use the `Market.USA` market: https://github.com/QuantConnect/Lean/pull/4328/files#diff-0417c0a6ef4ab67693f9e40a4a597914R201-R213

Reply:
Thanks for the thorough review! 👍 
We discussed this and avoided remapping it, because it would be rare if users specified a Market currently